backend/layers/processing/process_validate_atac.py

```diff
@@ -0,0 +1,224 @@
+        # fragment file to avoid uploading the same file multiple times
+        if self.business_logic.is_already_ingested(manifest.atac_fragment):
+            original_fragment_hash = self.hash_file(local_fragment_filename)
+        else:
+            # new fragment file is from a private source, so we must upload it
+            original_fragment_hash = None
+
+        # Validate the fragment with anndata file
+        try:
+            errors = self.schema_validator.validate_atac(local_fragment_filename, local_anndata_filename)
+        except Exception as e:
+            # for unexpected errors, log the exception and raise a ValidationAtacFailed exception
+            self.logger.exception("validation failed")
+            self.update_processing_status(
+                dataset_version_id, DatasetStatusKey.ATAC_FRAGMENT, DatasetConversionStatus.FAILED
+            )
+            raise ValidationAtacFailed(errors=[str(e)]) from None
+
+        if errors:
+            # if the validation fails, update the processing status and raise a ValidationAtacFailed exception
+            self.update_processing_status(
+                dataset_version_id, DatasetStatusKey.ATAC_FRAGMENT, DatasetConversionStatus.FAILED
+            )
+            raise ValidationAtacFailed(errors=errors)
+
+        # check to see if the new fragments is the same as the old fragment
+        # if it is the same, skip the upload and use link the old fragment to the new dataset
+        # This case is for when the dataset is reprocessed, or updated
+        if original_fragment_hash is not None and original_fragment_hash == self.hash_file(local_fragment_filename):
+            # get the artifact id of the old fragment, and add it to the new dataset
+            artifact_name = str(manifest.atac_fragment).split("/")[-1]
+            artifact = self.business_logic.database_provider.get_artifact_by_uri_suffix(artifact_name)
+            self.business_logic.database_provider.add_artifact_to_dataset_version(dataset_version_id, artifact.id)
+            # get the artifact id of the old fragment index, and add it to the new dataset
+            artifact = self.business_logic.database_provider.get_artifact_by_uri_suffix(artifact_name + ".tbi")
+            self.business_logic.database_provider.add_artifact_to_dataset_version(dataset_version_id, artifact.id)
+            self.update_processing_status(
+                dataset_version_id, DatasetStatusKey.ATAC_FRAGMENT, DatasetConversionStatus.UPLOADED
+            )
+            # TODO Copied would be a better status than uploaded
+        else:
+            fragment_artifact_id = self.create_atac_artifact(
+                local_fragment_filename,
+                DatasetArtifactType.ATAC_FRAGMENT,
+                dataset_version_id,
+                DatasetStatusKey.ATAC_FRAGMENT,
+                datasets_bucket,
+            )
+            self.create_atac_artifact(
+                local_fragment_filename + ".tbi",
```

Comment:
is this file created as part of `self.schema_validator.validate_atac(local_fragment_filename, local_anndata_filename)`? if so, could we make that more clear somehow? otherwise it seems like we're passing in a filepath to nothing